components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -84,11 +88,21 @@ public BrandingPreference addBrandingPreference(BrandingPreference brandingPrefe
 
         String resourceName = getResourceName
                 (brandingPreference.getType(), brandingPreference.getName(), brandingPreference.getLocale());
+        String resourceType = getResourceType(brandingPreference.getType());
         String tenantDomain = getTenantDomain();
         // Check whether a branding resource already exists with the same name in the particular tenant to be added.
-        if (isResourceExists(BRANDING_RESOURCE_TYPE, resourceName)) {
+        if (isResourceExists(resourceType, resourceName)) {
             throw handleClientException(ERROR_CODE_BRANDING_PREFERENCE_ALREADY_EXISTS, tenantDomain);
         }
+
+        if (APPLICATION_TYPE.equals(brandingPreference.getType())) {
+            // Check whether an application exists with the given name.
+            if (!isApplicationExists(brandingPreference.getName(), tenantDomain)) {
```

Comment:
Shall we improve this to match with the application ID scenario? https://github.com/wso2-enterprise/asgardeo-product/issues/16884#issuecomment-1886942357

Reply:
This `isApplicationExists` method was not implemented with this PR. Initially we planed to implement this application exist check in the second phase. So I think we can implement it seperately. WDYT?